Modules/_datetimemodule.c

```diff
@@ -1975,8 +1972,12 @@ wrap_strftime(PyObject *object, PyObject *format, PyObject *timetuple,
                     goto Done;
                 }
             }
-            ntoappend = sprintf(year_formatted, "%04ld", year_long);
-            ptoappend = year_formatted;
+
+            size_t buf_size = SIZEOF_LONG*5/2+2; /* maximum size of formatted year permitted by long */
+            char buf[buf_size];                  /* formatted year for %Y/%G */
```

Comment:
Ah yes using `sizeof` is definitely the better approach since the size is known at compile time. A macro seems unnecessary since it is unlikely to be reused anywhere else. Thanks again for the careful review!

Reply:
Hmm actually it's now failing the address sanitizer build test after I made the buffer size passed to `PyOS_snprintf` a compile-time value of `sizeof(buf)`, and it also failed with the macro approach as well. Can you help see what may be the cause?

Reply:
Ah found the issue. `buf` is assigned to `ptoappend`, which is used outside the local scope in which `buf` is declared. So unfortunately we can't declare `buf` near its apparent usage in this case. Have to move the declaration to the same scope as `ptoappend`.